### PR TITLE
Make sizeCalculator lazy && fix access level

### PR DESCRIPTION
**Pull request by zhongwuzw**

### CODE REVIEW

**Comment by SD10** [comment]
I’ll have to review this when I get back to the desktop @zhongwuzw. We shouldn’t mark the constraint class or any extensions as public. I believe good frameworks do not expose things that extend any part of Apple’s framework unless absolutely necessary 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by SD10** [comment]
It can lead to complications when a user has the same extension as we do. Developers often use similar extensions to abstract common tasks. We should also turn on the SwiftLint rule for explicit access control

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

File: Sources/Extensions/NSAttributedString+Extensions.swift [hunk 1]
```diff
@@ -26,15 +26,15 @@ import Foundation
 
 extension NSAttributedString {
 
-    func height(considering width: CGFloat) -> CGFloat {
+    public func height(considering width: CGFloat) -> CGFloat {
```
Comment (SD10) [inline Sources/Extensions/NSAttributedString+Extensions.swift]: These should be internal

File: Sources/Extensions/UIColor+Extensions.swift [hunk 2]
```diff
@@ -26,14 +26,14 @@ import Foundation
 
 extension UIColor {
 
-    static let incomingGray = UIColor(red: 230/255, green: 230/255, blue: 235/255, alpha: 1.0)
+    public static let incomingGray = UIColor(red: 230/255, green: 230/255, blue: 235/255, alpha: 1.0)
```
Comment (SD10) [inline Sources/Extensions/UIColor+Extensions.swift]: Debatable but let's leave these as internal for now

File: Sources/Extensions/UIView+Extensions.swift [hunk 3]
```diff
@@ -26,7 +26,7 @@ import UIKit
 
 extension UIView {
     
-    func fillSuperview() {
+    public func fillSuperview() {
```
Comment (SD10) [inline Sources/Extensions/UIView+Extensions.swift]: Internal for this class too 😓

File: Sources/Extensions/UIEdgeInsets+Extensions.swift [hunk 4]
```diff
@@ -26,11 +26,11 @@ import Foundation
 
 extension UIEdgeInsets {
 
-    var vertical: CGFloat {
+    public var vertical: CGFloat {
```
Comment (SD10) [inline Sources/Extensions/UIEdgeInsets+Extensions.swift]: Internal

File: Sources/Models/DetectorType.swift [hunk 5]
```diff
@@ -38,7 +38,7 @@ public enum DetectorType {
     //case hashtag
     //case custom
 
-    var textCheckingType: NSTextCheckingResult.CheckingType {
+    public var textCheckingType: NSTextCheckingResult.CheckingType {
```
Comment (SD10) [inline Sources/Models/DetectorType.swift]: Internal would be good for this too. I may have some future plans that could affect this

File: Sources/Models/NSConstraintLayoutSet.swift [hunk 6]
```diff
@@ -59,7 +59,7 @@ public class NSLayoutConstraintSet {
     ///
     /// - Returns: Self
     @discardableResult
-    func activate() -> Self {
+    public func activate() -> Self {
```
Comment (SD10) [inline Sources/Models/NSConstraintLayoutSet.swift]: To be honest, this class should not be public. MessageKit is not a constraint library. What to do about this considering it accidentally got exposed?

Comment (zhongwuzw) [reply Sources/Models/NSConstraintLayoutSet.swift]: No exposed currently, so let's make `internal`.

**Comment by zhongwuzw** [comment]
@SD10 , I updated `swiftlint`'s config.

>It can lead to complications when a user has the same extension as we do. Developers often use similar extensions to abstract common tasks. We should also turn on the SwiftLint rule for explicit access control

Emm, I think it's not lead to complications, extension belongs to different module, it would work safely. 😂

**Comment by SD10** [comment]
> Emm, I think it's not lead to complications, extension belongs to different module, it would work safely. 😂 

We’ve had complications at SwifterSwift with this problem. Module A extends UIKit with the same extension as Module B. The compiler can’t resolve which is which. I even tried things like `MessageKit.UIEdgeInsets` but even if that did work it would be very bad to work with. And in some situations you get a return type that is not prefixed by the module, ie. A UIKit method. So I basically never corrupt UIKit unless I’m forced to

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

File: Sources/Models/NSConstraintLayoutSet.swift [hunk 7]
```diff
@@ -24,18 +24,18 @@
 
 import UIKit
 
-public class NSLayoutConstraintSet {
+internal class NSLayoutConstraintSet {
```
Comment (SD10) [inline Sources/Models/NSConstraintLayoutSet.swift:27]: We just need to add an API breaking CHANGELOG entry under removed for this. Then we can merge

Comment (zhongwuzw) [reply Sources/Models/NSConstraintLayoutSet.swift:27]: 👌

**Pull request merged**